pwndbg/disasm/__init__.py

```diff
@@ -208,6 +209,43 @@ def get(address, instructions=1):
 }
 
 
+def can_run_first_emulate():
+    """
+    Disable the emulate config variable if we don't have enough memory to use it
+    See https://github.com/pwndbg/pwndbg/issues/1534
+    And https://github.com/unicorn-engine/unicorn/pull/1743
+    """
+    global first_time_emulate
+    if not first_time_emulate:
+        return True
+    first_time_emulate = False
+
+    try:
+        from mmap import mmap, MAP_ANON, MAP_PRIVATE, PROT_EXEC, PROT_READ, PROT_WRITE  # isort:skip
+
+        mm = mmap(  # novm
+            -1, 1024 * 1024 * 1024, MAP_PRIVATE | MAP_ANON, PROT_WRITE | PROT_READ | PROT_EXEC
+        )
+        mm.close()
```

Comment:
@disconnect3d 

I believe this should go to the `final` clause.

Reply:
Why? The allocation will only happen if there is no exception

Reply:
Oh you are right :)